init_milvus.py

```diff
@@ -0,0 +1,110 @@
+import os
+import time
+import glob
+from dotenv import load_dotenv
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from pymilvus import FieldSchema, CollectionSchema, DataType, utility, Collection, connections, model
+
+load_dotenv()
+
+
+def read_md_files(directory):
+    md_files = glob.glob(f"{directory}/**/*.md", recursive=True)
+    docs = []
+    for file in md_files:
+        with open(file, "r", encoding="utf-8") as f:
+            docs.append(f.read())
+    return docs
+
+def split_md_by_sections(content):
+    # I don't know if this is efficient or not, it leads to uneven chunking of sections and hence affects the search results\
+    # After some testing the search results seem to be better for recursive splitting
+    # headers_to_split_on = [
+    #     ("##", "Header 2"),
+    #     ("###", "Header 3"),
+    # ]
+
+    # markdown_splitter = MarkdownHeaderTextSplitter(headers_to_split_on=headers_to_split_on, strip_headers=False)
+    # md_header_splits = markdown_splitter.split_text(content)
+    text_splitter = RecursiveCharacterTextSplitter(chunk_size=1024, chunk_overlap=128, length_function=len, is_separator_regex=False,)
+    
+    splits = text_splitter.split_text(content)
+    return splits
+
+if __name__ == "__main__":
+    openai_key = os.getenv("OPENAI_API_KEY")
+    openai_ef = model.dense.OpenAIEmbeddingFunction(
+        model_name="text-embedding-ada-002", 
+        api_key=openai_key
+    )
+
+    # Read markdown files from the repo
+    repo_path = "/app/repo"
+    docs_folders = ["docs", "adaptors"]
+    corpus = []
+    for folder in docs_folders:
+        full_path = os.path.join(repo_path, folder)
+        md_files_content = read_md_files(full_path)
+        for content in md_files_content:
+            sections = split_md_by_sections(content)
+            corpus.extend(sections)
+
+    # Embed the corpus
+    embeddings = openai_ef.encode_documents(corpus)
+
+    # Connect to Milvus
+    milvus_uri = os.getenv('MILVUS_URI')
+    token = os.getenv('MILVUS_TOKEN')
+    print(f"Connecting to DB: {milvus_uri}")
+    connections.connect("default", uri=milvus_uri, token=token, db_name="apollo_db")
+
+    # Check if the collection exists
+    collection_name = "apollo_sample"
```

Comment:
Yes, we can pass this as an argument to the script (I have done this for the docs_path). But passing it as an argument does not seem much useful at the moment. What do you think?

Reply:
I think it will be useful later.

Please declare this and any other CLI args the top of the file so that users a can work out the expected usage. It's probably worth a line of docs or two in the readme to explain how to manually run this file.

Reply:
Sure, I will do it.